### PR TITLE
DR-2054: store per-snapshot catalog metadata

### PR DESCRIPTION
**Pull request by pshapiro4broad**
Add a new database object with its own DAO to store a json metadata blob that's linked to a snapshot.

### CODE REVIEW

**Review by fboulnois — APPROVED** [review]
This looks great. The DB operations are clean and the tests are concise but thorough.

File: src/test/java/bio/terra/service/search/SnapshotSearchMetadataDaoTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@Category(Connected.class)
```
Comment (tcjiang9) [inline src/test/java/bio/terra/service/search/SnapshotSearchMetadataDaoTest.java:36]: Do we actually need connected tests here? Can we not test the same thing with unit tests that verify jdbcTemplate.update is called with the correct params, and trust the underlying db does what it's supposed to. If we want to verify whether we can connect to the database, monitors and metrics should tell us that.

**Review by tcjiang9 — CHANGES_REQUESTED** [review]
I don't see the value of connected tests, I've been actively trying to remove them, can we not test the same thing with unit tests that verify jdbcTemplate.update is called with the correct params, and trust the underlying db does what it's supposed to? If we want to verify whether we can connect to the database, monitors and metrics should tell us that.

**Comment by pshapiro4broad** [comment]
> I don't see the value of connected tests, I've been actively trying to remove them, can we not test the same thing with unit tests that verify jdbcTemplate.update is called with the correct params, and trust the underlying db does what it's supposed to? If we want to verify whether we can connect to the database, monitors and metrics should tell us that.

In many cases I'd agree, we want to be careful and not add any more connected tests unless it's necessary. In this case I believe it is necessary, because there's no other way to verify that the liquibase change is correct, and that the SQL commands in string literals are correct. If we used an ORM library, or some other system that auto-generated database tables and DAO bindings from our code, then I can see how having these tests wouldn't be useful.

I do think that any other test that relies on this class wouldn't need to be connected, and could use a mocked version of the DAO class.

**Comment by pshapiro4broad** [comment]
If there's a concern of the tests affecting our build times, all four tests take less than 2s when I run them locally.

**Review by myessail — APPROVED** [review]

**Review by tcjiang9 — APPROVED** [review]

**Pull request merged**